packages/vite/src/node/server/index.ts

```diff
@@ -382,17 +380,15 @@ export async function createServer(
 
   server.transformIndexHtml = createDevHtmlTransformFn(server)
 
-  exitProcess = async () => {
-    try {
-      await server.close()
-    } finally {
-      process.exit()
-    }
-  }
-
-  process.once('SIGTERM', exitProcess)
-
   if (!middlewareMode && process.env.CI !== 'true') {
+    exitProcess = async () => {
+      try {
+        await server.close()
+      } finally {
+        process.exit()
+      }
+    }
+    process.once('SIGTERM', exitProcess)
```

Comment:
I think this is a good change during middleware mode, but what is the reason to also ignore SIGTERM in CI?

Reply:
I'm confused too, but git blame points to https://github.com/vitejs/vite/pull/3659 and https://github.com/vitejs/vite/pull/4082

Reply:
But in that PRs, `SIGTERM` is still listened in CI, no?
If we want this PR to only affect the way `SIGTERM` is handled in `middlewareMode`, shouldn't the code be:
```js
  if (!middlewareMode) {
    exitProcess = async () => {
      try {
        await server.close()
      } finally {
        process.exit()
      }
    }
    process.once('SIGTERM', exitProcess)
    if (process.env.CI !== 'true') {
      process.stdin.on('end', exitProcess)
    }
  }
```

Reply:
Yeah I think you're right upon reading the PRs. I'll add the change!